src/components/Calculator.js

```diff
@@ -1,44 +1,59 @@
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(item) {
+    const funcBtn = item.target.textContent;
+    this.setState((e) => calculate(e, funcBtn));
   }
 
   render() {
+    const { total, next, operation } = this.state;
+
```

Comment:
- [ ] Let's improve the construction of our Calculator component by handling states with `useState` hook, which will be effective in managing state of the functional component. Our code should look like the following example.

```jsx
// ./src/components/Calculator.js

import React, { useState } from 'react';
import calculate from ../logic;

function Calculator() {
    const [total, setTotal] = useState(0);
    return {
      <div>
       // Calculator app 
      </div>
    };
}
```